src/libs/antares/scenarioGroupParser/ScenarioGroupParser.cpp

```diff
@@ -0,0 +1,88 @@
+// Copyright 2007-2025, RTE (https://www.rte-france.com)
+// See AUTHORS.txt
+// SPDX-License-Identifier: MPL-2.0
+// This file is part of Antares-Simulator,
+// Adequacy and Performance assessment for interconnected energy networks.
+//
+// Antares_Simulator is free software: you can redistribute it and/or modify
+// it under the terms of the Mozilla Public Licence 2.0 as published by
+// the Mozilla Foundation, either version 2 of the License, or
+// (at your option) any later version.
+//
+// Antares_Simulator is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// Mozilla Public Licence 2.0 for more details.
+//
+// You should have received a copy of the Mozilla Public Licence 2.0
+// along with Antares_Simulator. If not, see <https://opensource.org/license/mpl-2-0/>.
+
+#include "include/antares/scenarioGroupParser/ScenarioGroupParser.h"
+
+#include <ANTLRInputStream.h>
+#include <fmt/format.h>
+
+#include "antares/exception/RuntimeError.hpp"
+
+#include "scenarioBuilderExpression/ScenarioBuilderBaseVisitor.h"
+#include "scenarioBuilderExpression/ScenarioBuilderLexer.h"
+#include "scenarioBuilderExpression/ScenarioBuilderParser.h"
+
+namespace Antares
+{
+class ScenarioBuilderImplVisitor: public ScenarioBuilderBaseVisitor
+{
+public:
+    std::any visitRules(ScenarioBuilderParser::RulesContext* ctx) override
+    {
+        return visitLine(ctx->line(0));
+    }
+
+    std::any visitLine(ScenarioBuilderParser::LineContext* ctx) override
+    {
+        auto group = std::any_cast<std::string>(visitGroup(ctx->group()));
+        auto year = std::any_cast<int>(visitYear(ctx->year()));
+        auto chronicle = std::any_cast<int>(visitChronicle(ctx->chronicle()));
+        return ScenarioGroupParser::Line{.groupName = group, .year = year, .chronicle = chronicle};
+    }
+
+    std::any visitGroup(ScenarioBuilderParser::GroupContext* ctx) override
+    {
+        auto groupName = ctx->IDENTIFIER()->getText();
+        return groupName;
+    }
+
+    std::any visitYear(ScenarioBuilderParser::YearContext* ctx) override
+    {
+        auto yearText = ctx->getText();
+        int year = std::stoi(yearText);
```

Comment:
`std::stoi` may throw

- [std::invalid_argument](https://en.cppreference.com/w/cpp/error/invalid_argument.html) if no conversion could be performed.
- [std::out_of_range](https://en.cppreference.com/w/cpp/error/out_of_range.html) if the converted value would fall out of the range of the result type or if the underlying function ([std::strtol](https://en.cppreference.com/w/cpp/string/byte/strtol.html) or [std::strtoll](https://en.cppreference.com/w/cpp/string/byte/strtol.html)) sets [errno](https://en.cppreference.com/w/cpp/error/errno.html) to [ERANGE](https://en.cppreference.com/w/cpp/error/errno_macros.html).

see https://en.cppreference.com/w/cpp/string/basic_string/stol